### PR TITLE
Miawong/sc 55710/redesign how we display custom license fields

### PR DESCRIPTION
**Pull request by miaawong**
#### What this PR does / why we need it: Redesign of custom license fields, for better visibility. 
Loom: https://www.loom.com/share/dbccca64fa364c96904b5308ea0803a7

#### Which issue(s) this PR fixes: https://app.shortcut.com/replicated/story/55710/redesign-how-we-display-custom-license-fields-on-license-rows

Fixes # 

#### Does this PR introduce a user-facing change?

```release-note
Updates the UI for customer license entitlements to provide better visibility.
```

#### Does this PR require documentation?

NONE

### CODE REVIEW

File: web/src/components/apps/AppLicense.jsx [hunk 1]
```diff
@@ -314,228 +327,183 @@ class AppLicense extends Component {
...
-                    {message}
-                  </p>
-                )}
-                {appLicense?.lastSyncedAt && (
-                  <p className="u-fontWeight--bold u-fontSize--small u-textColor--header u-lineHeight--default u-marginTop--10">
-                    Last synced {Utilities.dateFromNow(appLicense.lastSyncedAt)}
-                  </p>
-                )}
               </div>
+              {size(appLicense?.entitlements) >= 5 && (
```
Comment (miaawong) [inline web/src/components/apps/AppLicense.jsx:463]: the dropdown to view more entitlements

File: web/src/components/apps/AppLicense.jsx [hunk 2]
```diff
@@ -314,228 +327,183 @@ class AppLicense extends Component {
...
+                  entitlements={appLicense?.entitlements}
+                  entitlementsToShow={this.state.entitlementsToShow}
+                  toggleHideDetails={this.toggleHideDetails}
+                  toggleShowDetails={this.toggleShowDetails}
+                />
+              ) : (
+                appLicense.entitlements.length < 5 && (
+                  <div style={{ marginTop: "15px" }}>
+                    <LicenseFields
+                      entitlements={appLicense?.entitlements}
```
Comment (miaawong) [inline web/src/components/apps/AppLicense.jsx:499]: the component that renders all the entitlements

File: web/src/components/apps/LicenseFields.tsx [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+  background: none;
+  border: none;
+  color: #007cbb;
+  cursor: pointer;
+  font-size: 12px;
+  padding-left: 0;
+`;
+
+type Entitlements = {
+  label: string;
```
Comment (miaawong) [inline web/src/components/apps/LicenseFields.tsx:30]: let me know if I'm missing any type checking

File: web/src/components/apps/LicenseFields.tsx [hunk 4]
```diff
@@ -0,0 +1,135 @@
+import React from "react";
+// @ts-ignore
```
Comment (rianfowler) [inline web/src/components/apps/LicenseFields.tsx:2]:
```suggestion
// TODO: add type checking support for styled components or add a global ignore 
// @ts-ignore
```

File: web/src/components/apps/LicenseFields.tsx [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+              {entitlement.title}:{" "}
+              <span
+                className={`u-fontWeight--bold ${
+                  isTextField && "u-fontFamily--monospace"
+                }`}
+              >
+                {" "}
+                {isBooleanField
+                  ? entitlement.value.toString()
+                  : entitlement.value}{" "}
```
Comment (rianfowler) [inline web/src/components/apps/LicenseFields.tsx:125]:
```suggestion
                {entitlement.value.toString() + " "}
```
could do something like this. strings have `toString()` on them.

File: web/src/components/apps/LicenseFields.tsx [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+  cursor: pointer;
+  font-size: 12px;
+  padding-left: 0;
+`;
+
+type Entitlements = {
+  label: string;
+  title: string;
+  value: string;
+  valueType: string;
```
Comment (rianfowler) [inline web/src/components/apps/LicenseFields.tsx:33]:
```suggestion
  valueType: 'Text' | 'Boolean';
```

Might be able to narrow this.

File: web/src/components/apps/LicenseFields.tsx [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+}: {
+  entitlements: Entitlements[];
+  toggleShowDetails: (title: string) => void;
+  toggleHideDetails: (title: string) => void;
+  entitlementsToShow: string[];
+}) => {
+  return (
+    <CustomerLicenseFields className="flex flexWrap--wrap">
+      {entitlements?.map((entitlement) => {
+        const currEntitlement = entitlementsToShow?.find(
```
Comment (rianfowler) [inline web/src/components/apps/LicenseFields.tsx:50]: might be able to rename this to something clearer. is this like `shownEntitlement` or `displayedEntitlement` or `currentlySelectedEntitlement`

**Review by rianfowler — APPROVED** [review]
LGTM thanks for adding some type definitions

File: web/src/components/apps/LicenseFields.tsx (hunk 6 above)
Comment (miaawong) [reply web/src/components/apps/LicenseFields.tsx]: thanks!

File: web/src/components/apps/LicenseFields.tsx (hunk 7 above)
Comment (miaawong) [reply web/src/components/apps/LicenseFields.tsx]: updated to displayedEntitlement

**Review by rianfowler — APPROVED** [review]

**Pull request merged**